### PR TITLE
[wip] Rackspace Neutron Security Groups; Closes #391

### LINKED ISSUE #391: Neutron Security Groups (Rackspace)

### PR DESCRIPTION
**Pull request by jrperritt**
This PR implements Neutron Security Groups for Rackspace. When merged, it will close #391

### CODE REVIEW

File: rackspace/networking/v2/security/doc.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+//
+// The basic characteristics of Neutron Security Groups are:
+//
+// For ingress traffic (to an instance)
+//  - Only traffic matched with security group rules are allowed.
+//  - When there is no rule defined, all traffic are dropped.
+//
+// For egress traffic (from an instance)
+//  - Only traffic matched with security group rules are allowed.
+//  - When there is no rule defined, all egress traffic are dropped.
```
Comment (jamiehannaford) [inline rackspace/networking/v2/security/doc.go]: should it be "traffic are dropped" or "traffic is dropped"?

**Comment by jamiehannaford** [comment]
:+1: LGTM

File: rackspace/networking/v2/ports/delegate.go [hunk 2]
```diff
@@ -24,6 +24,9 @@ func Get(c *gophercloud.ServiceClient, networkID string) os.GetResult {
 
 // Create accepts a CreateOpts struct and creates a new network using the values
 // provided. You must remember to provide a NetworkID value.
+//
+// NOTE: Currently the SecurityGroup option is not implemented to work with
+// Rackspace.
```
Comment (smashwilson) [inline rackspace/networking/v2/ports/delegate.go]: I wish we had a way to log a warning if users tried to use it, for example. Logging from a library seems like a bad idea, though.

**Comment by smashwilson** [comment]
Looks reasonable to me!

File: rackspace/networking/v2/security/doc.go (hunk 1 above)
Comment (jrperritt) [inline rackspace/networking/v2/security/doc.go]: Haha, I'm not sure; they're your words ;)  I think it should be 'is'. I've changed it there and in the OpenStack Neutron `security` doc.go file.

File: rackspace/networking/v2/ports/delegate.go (hunk 2 above)
Comment (jrperritt) [inline rackspace/networking/v2/ports/delegate.go]: Yeah, I know, not great. This is another reason we need a completely well-posed schema from which to build the API.

**Comment by jrperritt** [comment]
Thank you for the review @smashwilson and @jamiehannaford   :smile:

**Pull request merged**